src/test/java/com/example/demo/service/PatientServiceTest.java

```diff
@@ -0,0 +1,380 @@
+package com.example.demo.service;
+
+import com.example.demo.exception.patient.PatientAlreadyExistsException;
+import com.example.demo.exception.patient.PatientIllegalDataException;
+import com.example.demo.exception.patient.PatientNotFoundException;
+import com.example.demo.mapper.PatientMapper;
+import com.example.demo.model.PageableContentDto;
+import com.example.demo.model.patient.FullPatientDataDTO;
+import com.example.demo.model.patient.Patient;
+import com.example.demo.model.patient.PatientDTO;
+import com.example.demo.repository.PatientRepository;
+import com.example.demo.repository.VisitRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mapstruct.factory.Mappers;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class PatientServiceTest {
+    private PatientRepository patientRepository;
+    private PatientService patientService;
+
+    @BeforeEach
+    void setUp() {
+        patientRepository = mock(PatientRepository.class);
+        PatientMapper patientMapper = Mappers.getMapper(PatientMapper.class);
+        VisitRepository visitRepository = mock(VisitRepository.class);
+        patientService = new PatientService(patientRepository, patientMapper, visitRepository);
+    }
+
+    @Test
+    void getAllPatients_ThereAreNoPatients_ReturnsEmptyPageableContentDto() {
+        //given
+        Pageable pageable = PageRequest.of(0, 10);
+        when(patientRepository.findAll(pageable)).thenReturn(Page.empty());
+        //when
+        PageableContentDto<PatientDTO> result = patientService.getAllPatients(pageable);
+        //then
+        assertEquals(0, result.pageNumber());
+        assertEquals(0, result.totalEntries());
+        assertEquals(1, result.totalNumberOfPages());
+        assertTrue(result.content().isEmpty());
+    }
+
+    @Test
+    void getAllPatients_ThereArePatients_ReturnsCorrectPageableContentDto() {
+        //given
+        Pageable pageable = PageRequest.of(0, 10);
+        Patient currentPatient = buildPatient();
+        List<Patient> patients = new ArrayList<>();
```

Comment:
mógłbyś użyc List.of(currentPatient) a jeszcze ładniej jak tworzysz liste jednoelementowa byłoby użyć:
Collections.singletonList(currentPatient)